### ISSUE TITLE
Thinning the dependencies

### ISSUE THREAD

**Issue by kaanaksit** [kaanaksit/odak#25]
I want to drop dependencies as much as possible. The current list of dependencies can be found in [here](https://github.com/kunguz/odak/blob/master/requirements.txt). This issue will track the progress towards dropping dependencies.

**Comment by kunguz** [comment]
To remove `scipy` from `requirements.txt`: 
https://github.com/kunguz/odak/blob/d3ac9c90a4769b537729a38e061ae2c666ee4f6f/requirements.txt#L6

All needed is to replace this line:
https://github.com/kunguz/odak/blob/e8de761740c73093789e7f721b282fdcd0940703/odak/tools/file.py#L13

**Comment by oliland** [comment]
We should also drop `requirements.txt` and specify loose requirements like `torch` in `install_requires`. See

https://packaging.python.org/discussions/install-requires-vs-requirements/

tl;dr: applications should pin packages explicitly in `requirements.txt`, libraries should specify loose requirements in `setup.py/install_requires`.

**Comment by kunguz** [comment]
In the current `setup.py`, `install_requires` is deeply connected to `requirements.txt`: 
https://github.com/kunguz/odak/blob/c626d4024a314780a62cb71218e534d2a3ce7519/setup.py#L7-L12

At the bare minimum, `torch` and `numpy` are everything `odak` needs truly. These two dependencies can be the most abstract ones. My fear in keeping an abstract `install_requires` lies in the fact that a user may want to use these other definitions that rely on the missing dependencies and can easily conclude **things not working**. Given that the user base of `odak` is/will be a diverse set of people, they may not have the experience to judge and conclude, **Oh, this is a missing dependency issue, let me pip super quick**.

**Comment by kunguz** [comment]
`requirements.txt` is also used by GitHub to map dependencies across repositories. Their statistics on dependencies and dependents or things like `used by` feature require `requirements.txt` to be available at a repository.

![image](https://user-images.githubusercontent.com/191201/135727116-db90549d-c3d1-4dd5-a043-f2982d1380c5.png)

**Comment by kunguz** [comment]
8f24ef2f707cf00f71c0327ccd87e4c93d19926c
Removes `scipy` as a dependency. In the next release, `0.2.0`, pip version won't be needing `scipy`.
